services/evm/executor.go

```diff
@@ -0,0 +1,196 @@
+package evm
+
+import (
+	"fmt"
+
+	"github.com/onflow/atree"
+	"github.com/onflow/flow-go/fvm/evm"
+	"github.com/onflow/flow-go/fvm/evm/emulator"
+	"github.com/onflow/flow-go/fvm/evm/emulator/state"
+	"github.com/onflow/flow-go/fvm/evm/precompiles"
+	"github.com/onflow/flow-go/fvm/evm/types"
+	flowGo "github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/go-ethereum/common"
+	gethTypes "github.com/onflow/go-ethereum/core/types"
+	"github.com/onflow/go-ethereum/eth/tracers"
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-evm-gateway/models"
+	"github.com/onflow/flow-evm-gateway/storage"
+)
+
+type BlockExecutor struct {
+	types.StateDB // todo change to types.ReadOnlyView
+	emulator      types.Emulator
+	chainID       flowGo.ChainID
+	block         *models.Block
+	blocks        storage.BlockIndexer
+	logger        zerolog.Logger
+	receipts      storage.ReceiptIndexer
+
+	// block dynamic data
+	txIndex uint
+	gasUsed uint64
```

Comment:
Can I get some more explanation on this and how it relates to `Call`

Reply:
These are needed when we are re-executing all or a portion of transactions in a block. Necessary for generating traces for a single transaction, or for all transactions in a given block. These are updated in memory, and passed in `types.BlockContext`, so that the traces are deterministic.